### ISSUE TITLE
Issue with RequestTimeout() after update from 2.2 to 5.3.1

### ISSUE THREAD

**Issue by Ashutor** [ggravlingen/pytradfri#156]
### Where are you using pytradfri (eg stand-alone, Home Assistant etc)
Using pytradfri a stand-alone with a few scripts. 

I tried to update pytradfri from a former version to the current version to use some of the newly implemented features. After the update I have not been getting various different errors from the libcoap_api file. 

### Version of pytradfri
Former version: 2.2 (I think)
Current version 5.3.1

### Expected behaviour
Scenario 1) Generation of PSK should complete and write to conf-file. Currently I get a timeout.
Scenaio 2) When calling "devices_commands = api(devices_command)" I get a TypeError.

### Actual behaviour
Scenario 1)
When running the test example example_sync.py, I get the below error:

> Please provide the 'Security Code' on the back of your Tradfri gateway: *********
Traceback (most recent call last):
  File "test2.py", line 101, in <module>
    run()
  File "test2.py", line 81, in run
    psk = api_factory.generate_psk(args.key)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 160, in generate_psk
    self._psk = self.request(Gateway().generate_psk(existing_psk_id))
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 92, in request
    return self._execute(api_commands)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 81, in _execute
    raise RequestTimeout() from None
pytradfri.error.RequestTimeout

Scenario 2)
If I try to use my former code that worked with version 2.2 (I think) without genering the psk and just calling "api_factory = APIFactory("IP", "Security Code")" then I get the following. I dont know if this is even a valid call anymore, but it does not complain at this point. It is then the api(devices_command) that throws the error (I am not that experienced in coding, so I dont know where to look if this method is still supported.) See code and error below:

>>> api_factory = APIFactory("192.168.1.201", "xxxxxxxxx")
>>> api = api_factory.request
>>> gateway = Gateway()
>>> devices_command = gateway.get_devices()
>>> devices_commands = api(devices_command)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 92, in request
    return self._execute(api_commands)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 79, in _execute
    return_value = subprocess.check_output(command, **kwargs)
  File "/usr/lib/python3.5/subprocess.py", line 316, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.5/subprocess.py", line 383, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.5/subprocess.py", line 676, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.5/subprocess.py", line 1221, in _execute_child
    restore_signals, start_new_session, preexec_fn)
TypeError: Can't convert 'NoneType' object to str implicitly

I have tried updating/reinstalling libcoap after the pytradfri update to ensure it was up-to-date. 
The dependencies for aiocoap (v0.4a1), DTLSSocket (v0.1.4) and typing (v3.6.4) are installed. 

Any help and guidance is much appreciated.

**Comment by ggravlingen** [comment]
@Ashutor you state that you run the example files. In your logs above, there is mention of `File "test2.py", line 101`. Did you rename the example files? It shouldn't matter but I'm trying to tye the loose ends. Also, what happens if you run the example_async-file?

**Comment by Ashutor** [comment]
@ggravlingen Thanks for assisting me in (hopefully) getting it back to a working state. 

I were just about to write a really long response in everything I have tried since I posted earlier today together with a new traceback from running the example_async file. This issue has been driving me nuts the since Friday evening where I updated - Last element I wanted to check was if - for some reason - the IP of the GW should have changed, and it did - I were trying to connect to nothing - Embarrassing... Tried with the correct / now updated / IP, and it works like a charm with the example_sync file - Now I just need to update my older scripts to use the updated functionality.

Thanks for a great package and sorry for wasting your time with such stupidity. 👍
